modules/giapi/src/main/scala/giapi/client/GiapiStatusDb.scala

```diff
@@ -0,0 +1,119 @@
+// Copyright (c) 2016-2018 Association of Universities for Research in Astronomy, Inc. (AURA)
+// For license information see LICENSE or https://opensource.org/licenses/BSD-3-Clause
+
+package giapi.client
+
+import cats.Applicative
+import cats.implicits._
+import cats.effect.Sync
+import cats.effect.Resource
+import cats.effect.Effect
+import cats.effect.ConcurrentEffect
+import cats.effect.implicits._
+import edu.gemini.aspen.giapi.status.StatusHandler
+import edu.gemini.aspen.giapi.status.StatusItem
+import edu.gemini.aspen.giapi.statusservice.StatusHandlerAggregate
+import edu.gemini.jms.activemq.provider.ActiveMQJmsProvider
+import fs2.concurrent.Queue
+import fs2.Stream
+
+/////////////////////////////////////////////////////////////////
+// Links status streaming with the giapi db
+/////////////////////////////////////////////////////////////////
+trait GiapiStatusDb[F[_]] {
+  def value(i: String): F[Option[StatusValue]]
+  def close: F[Unit]
+}
+
+object GiapiStatusDb {
+  private def dbUpdate[F[_]: Applicative, A](db:   GiapiDb[F],
+                                             name: String,
+                                             a:    A): F[Unit] =
+    a match {
+      case a: Int =>
+        db.update[Int](name, a)
+      case a: String =>
+        db.update[String](name, a)
+      case a: Float =>
+        db.update[Float](name, a)
+      case a: Double =>
+        db.update[Double](name, a)
+      case _ =>
+        Applicative[F].unit
+    }
+
+  def streamItemsToDb[F[_]: ConcurrentEffect](agg:   StatusHandlerAggregate,
+                                              db:    GiapiDb[F],
+                                              items: List[String]): F[Unit] = {
+    def statusHandler(q: Queue[F, (String, _)]) = new StatusHandler {
+
+      override def update[B](item: StatusItem[B]): Unit =
+        // Check the item name and enqueue it
+        if (items.contains(item.getName)) {
+          Option(item.getValue).foreach { a =>
+            q.enqueue1((item.getName, a)).toIO.unsafeRunAsync(_ => ())
+          }
+        }
+
+      override def getName: String = "Giapi staus db"
+    }
+
+    // A trivial resource that binds and unbinds a status handler.
+    def bind(q: Queue[F, (String, _)]): Resource[F, StatusHandler] =
+      Resource.make(
+        Effect[F].delay {
+          val sh = statusHandler(q)
+          agg.bindStatusHandler(sh)
+          sh
+        }
+      )(sh => Effect[F].delay(agg.unbindStatusHandler(sh)))
+
+    // Create a queue and put updates to forward them to the db
+    val s = for {
+      q <- Stream.eval(Queue.unbounded[F, (String, _)])
```

Comment:
Same comment here, `(String, Any)` might express the situation better than using an existential, since we're relying on it getting crushed to `Any` anyway.